### PR TITLE
Remove raw types warnings with Class

### PR DESCRIPTION
**Pull request by Emkas**

### CODE REVIEW

**Comment by Emkas** [comment]
Hmm... this is odd. Build failed. I use `mvn clean install` and all looks fine. Now I even did what fails in a build `mvn --no-transfer-progress install -Pstaging` and it ends fine as well. I there any command I should look for?

BTW. Shouldn't this be `mvn clean --no-transfer-progress install -Pstaging`?

**Comment by BalusC** [comment]
This PR is too big to quickly glance over, but I see that this change also affected the public API at a handful places (i.e. these changes basically also affect the generated javadocs). I recommend to not do that in the context of this PR.

**Comment by Emkas** [comment]
OK. PR is still big. If you want to to split, please tell me.

I removed everything what is touching public methods/fields etc.

File: impl/src/main/java/com/sun/faces/cdi/CdiProducer.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
 
     private String id = this.getClass().getName();
     private String name;
-    private Class<?> beanClass = Object.class;
+	private Class<?> beanClass = Object.class;
```
Comment (BalusC) [inline impl/src/main/java/com/sun/faces/cdi/CdiProducer.java:52]: Tab needs to be spaces. Also in rest of PR.

Comment (Emkas) [reply impl/src/main/java/com/sun/faces/cdi/CdiProducer.java:52]: There were only 2. Nevertheless, sorry for those remainings.

**Pull request merged**